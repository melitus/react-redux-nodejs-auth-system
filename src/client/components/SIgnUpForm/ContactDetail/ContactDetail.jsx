// @flow
import React from 'react';
import { Field, FormSection } from 'redux-form';

import { Input } from '../../Field/index';

const ContactDetail = ({ email, phoneNumber }) => (
  <FormSection name="contactDetail">
    <label target="signup.contactDetail.email">Email</label>
    <Field
      id="signup.personInfo.email"
      name="email"
      component={Input}
      placeholder="Email"
      type="email"
      value={email}
    />

    <label target="signup.contactDetail.phoneNumber">Phone Number</label>
    <Field
      id="signup.contactDetail.phoneNumber"
      name="phoneNumber"
      component={Input}
      placeholder="Phone Number"
      type="tel"
      value={phoneNumber}
    />
  </FormSection>
);

ContactDetail.defaultProps = {
  email: '',
  phoneNumber: '',
};

export default ContactDetail;
