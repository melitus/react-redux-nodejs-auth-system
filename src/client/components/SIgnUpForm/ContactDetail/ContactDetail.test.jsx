import React from 'react';
import enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15.4';

import ContactDetail from './ContactDetail';

enzyme.configure({ adapter: new Adapter() });

describe('<ContactDetail />', () => {
  it('should render default snapshot', () => {
    const wrapper = shallow(
      <ContactDetail />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
