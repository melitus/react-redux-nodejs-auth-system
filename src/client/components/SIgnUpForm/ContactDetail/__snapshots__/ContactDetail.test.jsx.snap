exports[`<ContactDetail /> should render default snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContactDetail
    email=""
    phoneNumber="" />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <label
          target="signup.contactDetail.email">
          Email
</label>,
        <Field
          component={undefined}
          id="signup.personInfo.email"
          name="email"
          placeholder="Email"
          type="email"
          value="" />,
        <label
          target="signup.contactDetail.phoneNumber">
          Phone Number
</label>,
        <Field
          component={undefined}
          id="signup.contactDetail.phoneNumber"
          name="phoneNumber"
          placeholder="Phone Number"
          type="tel"
          value="" />,
      ],
      "component": "div",
      "name": "contactDetail",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Email",
          "target": "signup.contactDetail.email",
        },
        "ref": null,
        "rendered": "Email",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": undefined,
          "id": "signup.personInfo.email",
          "name": "email",
          "placeholder": "Email",
          "type": "email",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Phone Number",
          "target": "signup.contactDetail.phoneNumber",
        },
        "ref": null,
        "rendered": "Phone Number",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": undefined,
          "id": "signup.contactDetail.phoneNumber",
          "name": "phoneNumber",
          "placeholder": "Phone Number",
          "type": "tel",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <label
            target="signup.contactDetail.email">
            Email
</label>,
          <Field
            component={undefined}
            id="signup.personInfo.email"
            name="email"
            placeholder="Email"
            type="email"
            value="" />,
          <label
            target="signup.contactDetail.phoneNumber">
            Phone Number
</label>,
          <Field
            component={undefined}
            id="signup.contactDetail.phoneNumber"
            name="phoneNumber"
            placeholder="Phone Number"
            type="tel"
            value="" />,
        ],
        "component": "div",
        "name": "contactDetail",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Email",
            "target": "signup.contactDetail.email",
          },
          "ref": null,
          "rendered": "Email",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": undefined,
            "id": "signup.personInfo.email",
            "name": "email",
            "placeholder": "Email",
            "type": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Phone Number",
            "target": "signup.contactDetail.phoneNumber",
          },
          "ref": null,
          "rendered": "Phone Number",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": undefined,
            "id": "signup.contactDetail.phoneNumber",
            "name": "phoneNumber",
            "placeholder": "Phone Number",
            "type": "tel",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
